File: office-utils/src/main/scala/io/github/ppdzm/utils/office/excel/workbook/OLE2WorkBook.scala
package io.github.ppdzm.utils.office.excel.workbook

import java.io.FileInputStream

import io.github.ppdzm.utils.office.excel.sheet.OLE2Sheet
import org.apache.poi.hssf.usermodel.HSSFWorkbook

/**
 * @author Created by Stuart Alex on 2019/3/29
 */
case class OLE2WorkBook(excelFileName: String, private val createWhenNotExist: Boolean = true) extends WorkBook {
    this.checkExtension("xls")
    this.backup(createWhenNotExist)
    override val workbook: HSSFWorkbook = {
        if (excelFile.exists()) {
            new HSSFWorkbook(new FileInputStream(excelFileName))
        } else {
            new HSSFWorkbook()
        }
    }

    /**
     * 获取sheet
     *
     * @param sheetName sheet名称
     * @param overwrite 复写
     * @return
     */
    def getSheet(sheetName: String, overwrite: Boolean = false): OLE2Sheet = {
        OLE2Sheet(this.workbook, sheetName, overwrite)
    }

    /**
     * 写sheet
     *
     * @param sheetName sheet名称
     * @param overwrite 复写
     * @param columns   列标题
     * @param rows      行数据
     * @return
     */
    def writeSheet(sheetName: String, overwrite: Boolean, columns: List[String] = List[String](), rows: List[List[Any]]): this.type = {
        try {
            OLE2Sheet(this.workbook, sheetName, overwrite)
                .writeColumnHeader(columns)
                .writeData(rows)
        } catch {
            case e: Exception =>
                this.close()
                throw e
        }
        this
    }

}

File: office-utils/src/main/java/io/github/ppdzm/utils/office/excel/workbook/OLE2WorkBook.java
package io.github.ppdzm.utils.office.excel.workbook;

import org.apache.poi.hssf.usermodel.HSSFWorkbook;

import java.io.FileInputStream;

/**
 * @author Created by Stuart Alex on 2023/5/16.
 */
public class OLE2WorkBook extends AbstractWorkBook {

    public OLE2WorkBook(String excelFileName) throws Exception {
        super(excelFileName);
        init(excelFileName, false, true);
    }

    public OLE2WorkBook(String excelFileName, boolean needBackup, boolean createWhenNotExist) throws Exception {
        super(excelFileName);
        init(excelFileName, needBackup, createWhenNotExist);
    }

    private void init(String excelFileName, boolean needBackup, boolean createWhenNotExist) throws Exception {
        checkExtension("xls");
        backup(needBackup, createWhenNotExist);
        if (excelFile.exists()) {
            workbook = new HSSFWorkbook(new FileInputStream(excelFileName));
        } else {
            workbook = new HSSFWorkbook();
        }
    }



}
